util/project_allocation_mapper.py

```diff
@@ -402,26 +402,6 @@ def contact_pi_via_rt(self, data):
         setattr(alloc, "decision_summary", f"RT ticket created with id: {ticket_id}")
         alloc.save()
 
-    def _update_user_membership(self, tas_project, user_ref, action=None):
```

Comment:
I'm starting to remove functions from this big mapper module, because the mapper should go away soon. It was only relevant when we were in this in-between step during user federation migration. We have some more tech debt to address here, notably the fact that we still use TAS project representations internally everywhere 😭